clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -672,6 +692,28 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         shard->last_window_ = memref.marker.marker_value;
     }
 
+    if (memref.marker.type == TRACE_TYPE_MARKER &&
+        memref.marker.marker_type == TRACE_MARKER_TYPE_BRANCH_TARGET) {
+        shard->last_branch_marker_value_ = memref.marker.marker_value;
+    }
+    if (knob_offline_ && shard->version_ >= TRACE_ENTRY_VERSION_BRANCH_INFO &&
+        type_is_instr_branch(memref.instr.type) &&
+        // I-filtered traces don't mark branch targets.
+        !TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
+                 shard->file_type_)) {
+        report_if_false(shard, memref.instr.type != TRACE_TYPE_INSTR_CONDITIONAL_JUMP,
+                        "The CONDITIONAL_JUMP type is deprecated and should not appear");
+        if (!type_is_instr_direct_branch(memref.instr.type)) {
+            shard->last_indirect_target_ = shard->last_branch_marker_value_;
+            report_if_false(shard,
+                            shard->last_indirect_target_ != 0 &&
+                                shard->prev_entry_.marker.type == TRACE_TYPE_MARKER &&
```

Comment:
Perhaps we can avoid last_branch_marker_value_ altogether and check the marker_value here itself.

Reply:
That is what I had originally and it broke on back-to-back indirect branches.

Reply:
Okay, add a comment if not there already.

Reply:
That's the comment in the header for this var, which I updated to try to make clearer.